### PR TITLE
minor: sevntu-checkstyle-sonar-plugin build with maven-checkstyle-plugin

### PR DESCRIPTION
**Pull request by vorburger**
with enforcement of Checkstyle's own Checkstyle rules, just like the
core sevntu-checks projects.  Mostly done just so that the
checkstyle-extensions.xml file, which gets modified for new checks in
sevntu-checks, has enforcement of the "no tabs" policy.

Signed-off-by: Michael Vorburger <vorburger@redhat.com>

### CODE REVIEW

File: sevntu-checkstyle-sonar-plugin/checkstyle.properties [hunk 1]
```diff
@@ -0,0 +1,4 @@
+checkstyle.suppressions.file=suppressions.xml
+checkstyle.header.file=https://raw.githubusercontent.com/checkstyle/checkstyle/master/config/java.header
```
Comment (rnveach) [inline sevntu-checkstyle-sonar-plugin/checkstyle.properties]: @romani Please confirm you are ok with specified header. Are we using the same header for everything we own?

File: sevntu-checkstyle-sonar-plugin/src/main/java/com/github/sevntu/checkstyle/sonar/CheckstyleExtensionPlugin.java [hunk 2]
```diff
@@ -1,14 +1,37 @@
...
 package com.github.sevntu.checkstyle.sonar;
 
 import java.util.Arrays;
 import java.util.List;
 
 import org.sonar.api.SonarPlugin;
 
+/**
+ * Sonar Plugin with {@link CheckstyleExtensionRepository} extension.
+ * @author unascribed
```
Comment (rnveach) [inline sevntu-checkstyle-sonar-plugin/src/main/java/com/github/sevntu/checkstyle/sonar/CheckstyleExtensionPlugin.java]: What is `unascribed`? Looks like first commiter was `rdiachenko`.
https://github.com/sevntu-checkstyle/sevntu.checkstyle/commits/master/sevntu-checkstyle-sonar-plugin/src/main/java/com/github/sevntu/checkstyle/sonar/CheckstyleExtensionPlugin.java

File: sevntu-checkstyle-sonar-plugin/src/main/java/com/github/sevntu/checkstyle/sonar/CheckstyleExtensionRepository.java [hunk 3]
```diff
@@ -8,32 +27,56 @@
 import org.sonar.api.rules.RuleRepository;
 import org.sonar.api.rules.XMLRuleParser;
 
+/**
+ * Sonar RuleRepository with Checkstyle Sevntu extensions.
+ * @author unascribed
```
Comment (rnveach) [inline sevntu-checkstyle-sonar-plugin/src/main/java/com/github/sevntu/checkstyle/sonar/CheckstyleExtensionRepository.java]: Another `unascribed`.

File: sevntu-checkstyle-sonar-plugin/import-control.xml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0"?>
+<!DOCTYPE import-control PUBLIC
+    "-//Puppy Crawl//DTD Import Control 1.2//EN"
+    "http://checkstyle.sourceforge.net/dtds/import_control_1_2.dtd">
+
+<import-control pkg="com.github.sevntu.checkstyle">
```
Comment (vorburger) [inline sevntu-checkstyle-sonar-plugin/import-control.xml]: OK, done. To be honest I didn't give this much thought - it's two just old classes.. ;-)

File: sevntu-checkstyle-sonar-plugin/import-control.xml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+  <allow pkg="java"/>
+  <allow pkg="org"/>
```
Comment (vorburger) [inline sevntu-checkstyle-sonar-plugin/import-control.xml]: OK, will do. To be honest I didn't give this much thought - it's two just old classes.. ;-)

File: sevntu-checkstyle-sonar-plugin/checkstyle.properties (hunk 1 above)
Comment (vorburger) [inline sevntu-checkstyle-sonar-plugin/checkstyle.properties]: @romani ?

File: sevntu-checkstyle-sonar-plugin/src/main/java/com/github/sevntu/checkstyle/sonar/CheckstyleExtensionPlugin.java (hunk 2 above)
Comment (vorburger) [inline sevntu-checkstyle-sonar-plugin/src/main/java/com/github/sevntu/checkstyle/sonar/CheckstyleExtensionPlugin.java]: FYI: unascribed is a buddy of mine. He appears in a number of classes as author in the JDK itself.. ;-)

OK, I'l change it to @rdiachenko

File: sevntu-checkstyle-sonar-plugin/src/main/java/com/github/sevntu/checkstyle/sonar/CheckstyleExtensionRepository.java (hunk 3 above)
Comment (vorburger) [inline sevntu-checkstyle-sonar-plugin/src/main/java/com/github/sevntu/checkstyle/sonar/CheckstyleExtensionRepository.java]: dito, done

**Comment by vorburger** [comment]
@rnveach re. "You did this change to manage checkstyle-extensions.xml but checkstyle-extensions.xml was not modified in this request, and travis is passing. I realize other PR has the changes, but it is not merged yet so this shouldn't pass right?" - well spotted!! I've investigated it, and figured that the maven-checkstyle-plugin parameter resourceIncludes has a stupid default of **/*.properties, so *.xml are not checked... good to know; [I'll just adjust the CS config of the project I'm using CS for with this important lesson learnt](https://github.com/opendaylight/odlparent/blob/master/odlparent/pom.xml#L1370) - thank you so much for noticing this here! Also added `<resourceIncludes>**/*.*</resourceIncludes>` to the pom.xml for sevntu-checkstyle-sonar-plugin in this Pull Request. Would you have the time and willingness to do this elsewhere, e.g. in sevntu-checks/pom.xml and checkstyle/pom.xml?

File: sevntu-checkstyle-sonar-plugin/checkstyle.properties (hunk 1 above)
Comment (romani) [inline sevntu-checkstyle-sonar-plugin/checkstyle.properties]: I am fine

**Pull request merged**